tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/Models/AzureCreateChatCompletionRequest.Serialization.cs

```diff
@@ -0,0 +1,241 @@
+﻿using OpenAI.Models;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+#nullable disable
+
+namespace AzureOpenAI.Models;
+
+internal partial class AzureCreateChatCompletionRequest : IJsonModel<AzureCreateChatCompletionRequest>
+{
+    AzureCreateChatCompletionRequest IJsonModel<AzureCreateChatCompletionRequest>.Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+    {
+        throw new NotImplementedException();
+    }
+
+    AzureCreateChatCompletionRequest IPersistableModel<AzureCreateChatCompletionRequest>.Create(BinaryData data, ModelReaderWriterOptions options)
+    {
+        throw new NotImplementedException();
+    }
+
+    string IPersistableModel<AzureCreateChatCompletionRequest>.GetFormatFromOptions(ModelReaderWriterOptions options)
+        => "J";
+
+    void IJsonModel<AzureCreateChatCompletionRequest>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
```

Comment:
likewise, I wonder if we could reuse the serialization implementation for the things that are the same and only add/modify the parts that are different.

Reply:
That seems likely.  My question on that would be -- what is the value of doing it?  I'm still wondering how we represent the added properties in the TypeSpec to indicate to the generator to generate just the diff from the base type.  Once we solve that piece , it'll be easier to say which approach would be less work for the generator.  If there's strong value to not duplicating generated serialization code, then we can prioritize figuring that part out.

Reply:
yeah, it's a good point that it will be easier on the generator to not have to understand such tricks. As far as the generator will be concerned is that there are two model types and two typespecs. It will generate serialization code as if there was no hierarchy,